### PR TITLE
Fetch image for video

### PR DESCRIPTION
**Pull request by drw86312**
Added a fetch image from URL condition in SBAsset's fetchImage method.

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -2,11 +2,12 @@ source 'https://github.com/CocoaPods/Specs.git'
 
 inhibit_all_warnings!
 
+    pod 'ReactiveCocoa'
+
 workspace 'CocoaBloc'
 xcodeproj './Projects/Library/CocoaBloc.xcodeproj'
 
 target :CocoaBloc do
-    pod 'ReactiveCocoa'
```
Comment (mglagola) [inline Podfile]: Why the move?

Comment (drw86312) [inline Podfile]: Oh ya forgot I did that. I was pod updating and getting the wrong Reactive Cocoa version. I changed it back.

**Comment by mglagola** [comment]
:+1:

File: Projects/Library/CocoaBloc/Camera/SBAsset.m [hunk 2]
```diff
@@ -138,6 +138,23 @@ - (RACSignal*) fetchImage {
             return nil;
         }];
     }
+
+    if (self.fileURL) {
+        return [RACSignal createSignal:^RACDisposable *(id<RACSubscriber> subscriber) {
+            AVAssetImageGenerator *generator = [[AVAssetImageGenerator alloc] initWithAsset:[[AVURLAsset alloc] initWithURL:self.fileURL options:nil]];
```
Comment (JRHeaton) [inline Projects/Library/CocoaBloc/Camera/SBAsset.m]: Need to reference self weakly from the block here. Also, for stylistic integrity, let's keep function brackets spaced from the end of the line. `- (void)foo {` vs `- (void)foo{`

File: Projects/Library/CocoaBloc/Camera/SBAsset.m [hunk 3]
```diff
@@ -138,6 +138,23 @@ - (RACSignal*) fetchImage {
...
+            generator.appliesPreferredTrackTransform = YES;
+            CMTime time = CMTimeMakeWithSeconds(0, 30);
+            [generator generateCGImagesAsynchronouslyForTimes:[NSArray arrayWithObject:[NSValue valueWithCMTime:time]] completionHandler:^(CMTime requestedTime, CGImageRef image, CMTime actualTime, AVAssetImageGeneratorResult result, NSError *error){
+                if (!error && result == AVAssetImageGeneratorSucceeded) {
+                    [subscriber sendNext:[UIImage imageWithCGImage:image]];
+                    [subscriber sendCompleted];
+                } else {
+                    [subscriber sendError:[NSError errorWithDomain:@"Cannot fetch image" code:101 userInfo:nil]];
```
Comment (JRHeaton) [inline Projects/Library/CocoaBloc/Camera/SBAsset.m]: Domain isn't description, for errors. Ideally, you will have a constant, reverse-DNS style domain string like we do for the CB API error domain, and use that here, inserting descriptions or information via `NSLocalizedDescriptionKey` or similar in the `userInfo` dictionary of the error.

**Pull request merged**